src/services/DataManagementService.ts

```diff
@@ -0,0 +1,22 @@
+import { getDataSource } from '@/db/data-source';
+
+export class DataManagementService {
+  /**
+   * Resets all application data by deleting all database entries
+   * @returns Promise<void>
+   */
+  static async resetAllData(): Promise<void> {
+    try {
+      const dataSource = getDataSource();
+      await dataSource.transaction(async (transactionManager) => {
+        await transactionManager.query('DELETE FROM grade');
+        await transactionManager.query('DELETE FROM exam');
+        await transactionManager.query('DELETE FROM subject');
+        await transactionManager.query('DELETE FROM school');
```

Comment:
```suggestion
        await transactionManager.query();
```

Make sure you properly understand how query keys work :)